text/0995-deprecate-non-colocated-components.md

```diff
@@ -0,0 +1,247 @@
+  components/
+    foo.js 
+    foo.hbs
+    namespace/
+      bar.js
+      bar.hbs
+```
+
+Or using `component-structure=nested`
+
+```
+{app,addon}/
+  components/
+    foo/
+      index.js 
+      index.hbs
+    namespace/
+      bar/
+        index.js
+        index.hbs
+```
+
+Note, however, that classi components _importing_ the `layout` and setting it on an `@ember/component` will still work.
+The key thing being deprecated is the runtime resolution of templates, so if there is an import involved, there is no runtime resolution.
+
+### `ember-source`
+
+The blueprint for components will need to remove the `classic` option:
+
+Here is the output at the time of authoring this RFC:
+```bash 
+❯ pnpm ember g component --help
+Requested ember-cli commands:
+
+ember generate <blueprint> <options...>
+  Generates new code from blueprints.
+  aliases: g
+  --dry-run (Boolean) (Default: false)
+    aliases: -d
+  --verbose (Boolean) (Default: false)
+    aliases: -v
+  --pod (Boolean) (Default: false)
+    aliases: -p, -pods
+  --classic (Boolean) (Default: false)
+    aliases: -c
+  --dummy (Boolean) (Default: false)
+    aliases: -dum, -id
+  --in-repo-addon (String) (Default: null)
+    aliases: --in-repo <value>, -ir <value>
+  --lint-fix (Boolean) (Default: true)
+  --in (String) (Default: null) Runs a blueprint against an in repo addon. A path is expected, relative to the root of the project.
+  --typescript (Boolean) Generates a version of the blueprint written in TypeScript (if available).
+    aliases: -ts
+
+      component <name> <options...>
+        Generates a component.
+        --path (String) (Default: components)
+          aliases: --no-path (--path="")
+        --component-class (@ember/component, @glimmer/component, @ember/component/template-only, "") (Default: --no-component-class)
+          aliases: -cc (--component-class=@ember/component), -gc (--component-class=@glimmer/component), -tc (--component-class=@ember/component/template-only), -nc (--component-class=""), --no-component-class (--component-class=""), --with-component-class (--component-class=@glimmer/component)
+        --component-structure (flat, nested, classic) (Default: flat)
+          aliases: -fs (--component-structure=flat), -ns (--component-structure=nested), -cs (--component-structure=classic)
+
+```
+
+Specifically, we'll remove:
+- `--component-structure=classic` and its alias
+
+
+## How We Teach This
+
+Guides and docs already don't mention the above old layouts.
+
+### How to migrate off old layouts 
+
+#### component templates in `templates/components`
+
+**Before**
+```
+{app,addon}
+  components/
+    demo.js
+  templates/
+    components/
+      demo.hbs
+```
+
+**After**
+```
+{app,addon}
+  components/
+    demo.js
+    demo.hbs
+```
+
+**After (gjs)**
+```
+{app,addon}
+  components/
+    demo.gjs
+```
+
+#### Pods
+
+
+**Before**
+```
+{app,addon}
+  components/
+    demo/
+      component.js
+      template.hbs
+```
+
+**After**
```

Comment:
I feel like we should not be recommending the "index" variant for people because of a few bugs around that right now. Sure we can make it work but I think we can just suggest the "non-nested" as the after version in the RFC 🤷

Reply:
index is _the_ transition path from component.js/template.hbs since Octane. It's a primary, first-class, feature. We can't just get rid of it (by omission) because there is a bug in the unstable embroider resolver (tho, I think that was fixed! (you fixed it! :muscle: ))

Reply:
We discussed in RFC review and agreed that showing index.js is OK because it's a supported node-ism that we need to continue to support, and any bugs are just bugs that we need to fix (in embroider).